src/commands/fun/dicksize.sakuria.ts

```diff
@@ -1,51 +1,130 @@
+          embeds: [createErrorEmbed("For you guys' healthiness, only 20 people can join the fight at once(including you).")]
+        };
+      }
+
+      const authorDickSize = randomDickSize();
+      const allDicks: [User, number][] = [[message.author, authorDickSize]];
+      const toFieldData = function([name, value]: string[]): { name: string, value: string, inline: true } {
+        return { name, value, inline: true }
+      };
+      // actually only used for embed image because there might be multiple
+      let winner: User = message.author;
+      let largestDickSize = authorDickSize;
+
+      mentionedUsers.forEach((user: User) => {
+        const dickSize = 3000;
+
+        if (dickSize > largestDickSize) {
+          winner = user;
+          largestDickSize = dickSize;
+        }
+        allDicks.push([user, dickSize]);
+      });
+
+      const dickAmountMod3 = allDicks.length % 3;
+      const embedTemplate = new SakuriaEmbed({
+        title: "The legendary battle of dongs",
+        thumbnail: winner.displayAvatarURL({
+          dynamic: true,
+          format: "png",
+          size: 128,
+        })
+      });
+
+      // 2000 not 1900 here because the maximum text in Embed is 2048
+      if (largestDickSize > 2000) {
+        return {
+          embeds: [
+            embedTemplate
+              .setDescription("This battle of the dongs is too much to just say as is, so here's the brief:\n")
+              .setFields([
+                ...allDicks
+                  .map(([user, dickSize]) =>
+                    [
+                      `${dickSize >= largestDickSize ? "(Winner) " : ""}${user.tag}`,
+                      `Dong size: ${dickSize.toString()}`
+                    ]
+                  )
+                  .map(toFieldData),
+                ...(dickAmountMod3 ? createInlineBlankField(dickAmountMod3) : [])
+              ])
+          ],
           files: [
             {
               name: "battle.txt",
               attachment: Readable.from(
-                `${message.author.username}'s long dong:\n 8${"=".repeat(dicksize)}D\n` +
-                  `${message.mentions.users.first()?.username}'s long dong:\n 8${"=".repeat(enemyDicksize)}D\n`
+                allDicks.reduce((acc, [user, dickSize]) =>
+                  `${user.tag}'s dong: 8${"=".repeat(dickSize)}D`, ""
+                )
               ),
             },
-          ],
-        };
-      } else
-        return `
-        8${"=".repeat(dicksize)}D ${dicksize}cm ${message.author}
-        8${"=".repeat(enemyDicksize)}D ${enemyDicksize}cm ${message.mentions.members.first()}
-        diff: ${Math.abs(dicksize - enemyDicksize)}cm
-        winner: ${dicksize > enemyDicksize ? message.author : message.mentions.members.first()}
-      `;
+          ]
+        }
+      } else {
+        return {
+          embeds: [
+            embedTemplate.setFields([
+              ...allDicks
+                .map(([user, dickSize]) =>

```

Comment:
you should probably turn this into a logic function or something, this seems pretty messy